docs/source/order.rst

```diff
@@ -0,0 +1,153 @@
+.. _order:
+
+Ordering
+========
+
+.. note::
+
+   This is an advanced topic that most users won't need to worry about.
+
+When Dask is given a task graph to compute, it needs to choose a an order to
+execute the tasks in. We have some constraints: dependencies must be executed
+before their dependants. But beyond that there's a large space of options. We
+want Dask to choose an ordering that maximizes parallelism while minimizing
+the footprint necessary to run a computation.
+
+At a high level, Dask has a policy that works towards *small goals* with *big steps*.
+
+1.  **Small goals**: prefer tasks that have few total dependents and whose final
+    dependents have few total dependencies.
+
+    We prefer to prioritize those tasks that help branches of computation that
+    can terminate quickly.
+
+    With more detail, we compute the total number of dependencies that each
+    task depends on (both its own dependencies, and the dependencies of its
+    dependencies, and so on), and then we choose those tasks that drive towards
+    results with a low number of total dependencies.  We choose to prioritize
+    tasks that work towards finishing shorter computations first.
+
+2.  **Big steps**: prefer tasks with many dependents
+
+    However, many tasks work towards the same final dependents.  Among those,
+    we choose those tasks with the most work left to do.  We want to finish
+    the larger portions of a sub-computation before we start on the smaller
+    ones.
+
+This is done with :func:`dask.order.order`. 
```

Comment:
This high-level discussion is great! I would suggest linking/consolidating with these two other docs:

- [Scheduling Policies](https://distributed.dask.org/en/latest/scheduling-policies.html) from the dask.distributed docs.
- [Scheduling in Depth](https://docs.dask.org/en/latest/scheduling-policy.html) from the dask docs (I'm not sure if this doc is linked anywhere, but it appears in search)